### ISSUE TITLE
Crashes on launch with GOG version of StS

### ISSUE THREAD

**Issue by artemiswkearney** [Tempus/Slay-the-Spire-Together#29]
```
Exception in thread "LWJGL Application" com.badlogic.gdx.utils.GdxRuntimeException: java.lang.UnsatisfiedLinkError: com.codedisaster.steamworks.SteamAPI.getSteamMatchmakingPointer()J
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication$1.run(LwjglApplication.java:133)
Caused by: java.lang.UnsatisfiedLinkError: com.codedisaster.steamworks.SteamAPI.getSteamMatchmakingPointer()J
	at com.codedisaster.steamworks.SteamAPI.getSteamMatchmakingPointer(Native Method)
	at com.codedisaster.steamworks.SteamMatchmaking.<init>(SteamMatchmaking.java:132)
	at chronoMods.steam.NetworkHelper.initialize(NetworkHelper.java:75)
	at chronoMods.TogetherManager.receivePostInitialize(TogetherManager.java:223)
	at basemod.BaseMod.publishPostInitialize(BaseMod.java:2240)
	at basemod.patches.com.megacrit.cardcrawl.core.CardCrawlGame.PostInitializeHook.Postfix(PostInitializeHook.java:9)
	at com.megacrit.cardcrawl.core.CardCrawlGame.create(CardCrawlGame.java:308)
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication.mainLoop(LwjglApplication.java:149)
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication$1.run(LwjglApplication.java:126)
```
I'd like to be able to play this with my partner, but not enough to buy the game 2 more times. I'm fine with a lack of matchmaking. Does the netcode completely rely on Steam?

**Comment by Tempus** [comment]
This mod requires Steam, sorry.

**Comment by artemiswkearney** [comment]
fucked up

**Comment by Tempus** [comment]
That's quite rude of you.

**Comment by artemiswkearney** [comment]
Why? I'm just saying it's sad that a bunch of people can't use what looks
like a really cool mod.

**Comment by Tempus** [comment]
I'm gonna give you the benefit of a doubt here, but telling someone the thing they spent 400+ hours of unpaid, high skill labour into is _fucked up_ is not exactly the height of being polite. 

There's all sorts of huge problems with supporting cross-platform networking, which is why most triple A companies don't do it, and the ones that do hire teams of dedicated network engineers to work on their backend servers and their netcode, which of course themselves cost big monthly fees. And then the implemented netcode is often unstable for months post-release, despite having full time paid teams maintaining it.

By contrast, I can pay 50$ to steam to get a Steamworks kit, use the existing steam auth to open a secure and obsfucated P2P channel, that's quite stable and costs me no per user or monthly fees. End users don't need any special knowledge or port forwarding or unique setups because Steam has spent decades making their netcode robust. All I have to do is manage the connection post-handoff.

As for mobile users, I won't be writing a jailbroken app hijacker to install this mod. I won't be writing a homebrew wrapper for the Switch to install this mod. I also won't be writing from scratch netcode and servers for GoG or other PC avenues.

Instead, those people can kindly pay 20$ _to someone who is not me_ to have anywhere from several to dozens of hours of great fun. 

Long story short, yes. Calling the work I've done fucked up because you have to fork out the price of a KFC 2 person family meal is pretty rude. Hate to unload this on you but I've been getting a lot of entitled people sending ugly messages to me, and I feel a little need to vent - please try to be kinder to others in the future.

**Comment by artemiswkearney** [comment]
I'm not saying the mod is fucked up. I'm saying it's fucked up *in the
abstract* that it's only available for one version of the game, and based
on your explanation, it looks like the specific thing that's fucked up is
that there's no cross-platform networking solution for Java games that's as
easy for you to use as Steamworks.
I'm a dev myself, and in the Unity world, there are multiple peer-to-peer
networking packages that have perfectly serviceable free tiers and don't
tether your game to one platform or device type. It's fucked up that you,
as a modder presumably making nothing from this project, had to pay Valve
$50 just to get decent netcode for your mod. And it's fucked up that this
creates an ecosystem where, if I want to fork the mod so I can use it, I'd
have to reverse-engineer Valve's whole API (or at least the wrapper you
used). Netcode isn't easy, but for something as asynchronous as this
(especially since it looks like you're rolling your own messages with only
41 different types), it feels like there ought to be something out there
that doesn't cost $50 and lock the mod to one platform.

**Comment by artemiswkearney** [comment]
Looked into it further. It looks like Discord provides a similar drop-in API, with Java bindings available [here](https://github.com/JnCrMx/discord-game-sdk4j). This would support all PC platforms without requiring users to purchase anything.
I'd of course be grateful if you'd implement it yourself, and quite possibly even willing to pay for the work of doing so. If you're not interested, though, would you consider adding a license so I can fork this repo and do it myself without violating IP laws?

**Comment by Tempus** [comment]
If you want to pay for features, I've got a [patreon here](https://www.patreon.com/chronometrics) where I vastly underprice myself but somehow still charge more than most of Patreon. If you'd prefer to fork and collab on it you can hop on the discord and give me a message. The code base has gotten a bit out of control and I rarely push commits (as you might expect from a solo dev hobby project), but it's only mostly spaghetti and I can be more diligent if others are working on it as well.
